java/src/org/openqa/selenium/grid/node/config/SessionCapabilitiesMutator.java

```diff
@@ -126,9 +127,11 @@ private Map<String, Object> mergeChromiumOptions(Map<String, Object> stereotypeO
         toReturn.put("extensions", stereotypeExtensions);
       }
 
-      if (name.equals("binary")) {
-        Object binary = capsOptions.get("binary");
-        toReturn.put("binary", binary);
+      // Grid will know the browser paths so preference is given to stereotype value in this case
+      // If the value is not present in stereotype but present in capabilities only then use capabilities value
+      if (name.equals("binary") && stereotypeOptions.get("binary") == null) {
+          Object binary = capsOptions.get("binary");
+          toReturn.put("binary", binary);
```

Comment:
I was thinking about this again and I think we should completely ignore `binary` when it comes in the session request capabilities. Ideally, the person who setup the Grid already has left all paths configured. 

This could also affect cloud vendors, probably BrowserStack sets the binary path where the browser is, IIRC Sauce does something like that. If a user sets the binary in a place where the browser is not present on the remote machine, session creation will fail.

Reply:
I will make the changes. I see your point. Grid should be the source of truth in such cases.